react/features/toolbar/components/PrimaryToolbar.web.js

```diff
@@ -0,0 +1,294 @@
+
+            acc.push(splitter);
+        }
+
+        const _onClick = button.onClick;
+        const _onMount = button.onMount;
+
+        acc.push(
+            <ToolbarButton
+                button = { button }
+                key = { key }
+                onClick = { _onClick }
+                onMount = { _onMount } />
+        );
+
+        return acc;
+    }
+
+    /**
+     * Sets on mount handlers for some buttons.
+     *
+     * @private
+     * @returns {void}
+     */
+    _setOnMountHandlers(): void {
+        const { onMountHandlers } = this.state;
+        const { _toolbarButtons } = this.props;
+
+        Object.keys(onMountHandlers).forEach(key => {
+            const button = _toolbarButtons.get(key);
+
+            if (button) {
+                button.onMount = onMountHandlers[key];
+                _toolbarButtons.set(key, button);
+            }
+        });
+    }
+}
+
+/**
+ * Maps part of Redux actions to component's props.
+ *
+ * @param {Function} dispatch - Redux action dispatcher.
+ * @returns {{
+ *      _onFullScreenToggled: Function,
+ *      _onMouseOut: Function,
+ *      _onMouseOver: Function,
+ *      _onShowDesktopSharingButton: Function
+ * }}
+ * @private
+ */
+function _mapDispatchToProps(dispatch: Function): Object {
+    return {
+        /**
+         * Dispatches an action signalling that full screen mode is toggled.
+         *
+         * @param {boolean} isFullScreen - Show whether fullscreen mode is on.
+         * @returns {Object} Dispatched action.
+         */
+        _onFullScreenToggled(isFullScreen: boolean) {
+            return dispatch(toggleFullScreen(isFullScreen));
+        },
+
+        /**
+         * Dispatches an action signalling that toolbar is not being hovered.
+         *
+         * @returns {Object} Dispatched action.
+         */
+        _onMouseOut() {
+            return dispatch(setToolbarHovered(false));
+        },
+
+        /**
+         * Dispatches an action signalling that toolbar is now being hovered.
+         *
+         * @returns {Object} Dispatched action.
+         */
+        _onMouseOver() {
+            return dispatch(setToolbarHovered(true));
+        },
+
+        /**
+         * Dispatches an action signalling that desktop sharing button
+         * should be shown.
+         *
+         * @returns {Object} Dispatched action.
+         */
+        _onShowDesktopSharingButton() {
+            dispatch(showDesktopSharingButton());
+        }
+    };
+}
+
+/**
+ * Maps part of Redux store to component's props.
+ *
+ * @param {Object} state - Redux store.
+ * @returns {{
+ *     _audioMuted: boolean,
+ *     _locked: boolean,
+ *     _toolbarButtons: Map,
+ *     _videoMuted: boolean
+ *}}
+ * @private
+ */
+function _mapStateToProps(state: Object): Object {
+    const { primaryToolbarButtons } = state['features/toolbar'];
+
+    return {
+        ...abstractMapStateToProps(state),
```

Comment:
abstractMapStateToProps properties don't seem to be used in this class or am I missing something?